docs/core/diagnostics/diagnostic-libraries.md

```diff
@@ -0,0 +1,161 @@
+- Builds the `IHost` instance.
+- Gets an instance of the <xref:Microsoft.Extensions.Hosting.IHostApplicationLifetime> interface from the service provider.
+- Gets an instance of the <xref:Microsoft.Extensions.Diagnostics.HealthChecks.HealthCheckService> class from the service provider.
+- Defines a delegate that will be invoked when the application starts, is stopping, and eventually stops—that requests the health check service to perform a health check and display the result.
+- Registers the delegate with the `ApplicationStarted`, `ApplicationStopping`, and `ApplicationStopped` events of the `IHostApplicationLifetime` instance.
+- Runs the application.
+
+### Resource utilization health checks
+
+To perform health checks on the resource utilization of your .NET apps, add a package reference to [Microsoft.Extensions.Diagnostics.HealthChecks.ResourceUtilization](https://www.nuget.org/packages/Microsoft.Extensions.Diagnostics.HealthChecks.ResourceUtilization). On an `IServiceCollection` instance, chain a call from <xref:Microsoft.Extensions.DependencyInjection.HealthCheckServiceCollectionExtensions.AddHealthChecks%2A> to <xref:Microsoft.Extensions.Diagnostics.HealthChecks.ResourceUtilizationHealthCheckExtensions.AddResourceUtilizationHealthCheck%2A>. The following example demonstrates how to use the `AddResourceUtilizationHealthCheck` extension method to add a resource utilization health check to an `IServiceCollection` instance:
+
+:::code source="snippets/health-checks/Program.cs":::
+
+The preceding code:
+
+- Creates a new <xref:Microsoft.Extensions.Hosting.HostApplicationBuilder> instance.
+- Adds resource monitoring by calling <xref:Microsoft.Extensions.Diagnostics.ResourceMonitoring.ResourceMonitoringExtensions.AddResourceMonitoring%2A>.
+- Adds a health check for resource utilization by chaining a call from the <xref:Microsoft.Extensions.DependencyInjection.HealthCheckServiceCollectionExtensions.AddHealthChecks%2A> call to the <xref:Microsoft.Extensions.Diagnostics.HealthChecks.ResourceUtilizationHealthCheckExtensions.AddResourceUtilizationHealthCheck%2A> extension method.
+- Builds the `IHost` instance.
+- Gets an instance of the <xref:Microsoft.Extensions.Diagnostics.HealthChecks.HealthCheckService> class from the service provider.
+- Performs a health check and displays the result.
+- Runs the application.
+
+> [!NOTE]
+> The `Microsoft.Extensions.Diagnostics.HealthChecks.ResourceUtilization` library assumes that the consumer will register the dependent call to <xref:Microsoft.Extensions.Diagnostics.ResourceMonitoring.ResourceMonitoringExtensions.AddResourceMonitoring%2A>. If you don't register this, when resolving the `HealthCheckService` an exception is thrown.
+
+## Kubernetes probes
+
+The [Microsoft.Extensions.Diagnostics.Probes](https://www.nuget.org/packages/Microsoft.Extensions.Diagnostics.Probes) NuGet package provides support for Kubernetes probes. This package is currently experimental and is subject to change.
```

Comment:
I would not mention here that package is experimental. We have the whole infra for informing developer about it, and mentioning it in docs is simple to forget and get out of sync with code.